Marlin/src/lcd/extui/lib/creality_dwin/creality_dwin.cpp

```diff
@@ -666,13 +704,19 @@ void CrealityDWINClass::Menu_Item_Handler(uint8_t menu, uint8_t item, bool draw/
           else {
             Popup_Handler(Home);
             gcode.process_subcommands_now_P(PSTR("G28"));
+            #if ENABLED(Z_SAFE_HOMING)
+              planner.synchronize();
+              char buf[20];
+              sprintf(buf, "G0 X%i Y%i", Z_SAFE_HOMING_X_POINT, Z_SAFE_HOMING_Y_POINT);
+              gcode.process_subcommands_now_P(buf);
+            #endif
```

Comment:
Why do you need to send the printer to its homing point here? If Z_SAFE_HOMING is enabled the print head should already be at this position.

Reply:
Not exactly: while homing with a bed probe, the extruder is moved such that the _probe_ ends up at (Z_SAFE_HOMING_X_POINT, Z_SAFE_HOMING_Y_POINT) and that's where it leaves it, meaning that the nozzle tip is left at (Z_SAFE_HOMING_X_POINT + NOZZLE_TO_PROBE_OFFSET.x, Z_SAFE_HOMING_Y_POINT + NOZZLE_TO_PROBE_OFFSET.y). This is very annoying for e.g. calibrating the z-offset after auto-homing, as you have to manually move to (Z_SAFE_HOMING_X_POINT, Z_SAFE_HOMING_Y_POINT), otherwise you will be changing the offset based on a point different from the one the Z-axis was homed with (not a big difference if your bed is ~level, but not right either!). In case a bed probe is not used for homing, this should indeed have no effect.

Reply:
That all makes sense, but I'd follow up with this; Why would you be using the auto home item and manually setting your z offset when there is a fully featured z offset menu a few menu items down from that? A menu which will not only handle homing and centering the nozzle, but will move it down to the bed for you and actively move it as you adjust the offset.

Reply:
That is a good question, I did use the menu to set the z-offset. I did not, however, use the "Home z-axis" item from that menu - with some further testing right now I found:
1. Maybe an issue: "Home z-axis" does not move to (Z_SAFE_HOMING_X_POINT, Z_SAFE_HOMING_Y_POINT) before homing, unless no homing has been performed and the full auto-home command is issued. This is not necessarily an issue if you deliberately want to set the z-offset based on a point you have manually chosen, but it likely is an issue if the homing point is random (wherever the extruder randomly happened to be positioned).
2. A serious issue: "Home z-axis" suffers from the same issue fixed in the change above - it homes the z-axis, but then leaves the nozzle wherever it ends up. Now you will be setting the z-offset using a point that was not the homing target. Ideally the same fix should be applied to the "Home z-axis" item, or perhaps exclusively to it (but that can cause another positioning inconsistency as in point 1)?

Reply:
Alright, so all things considered I'm going to exclusively fix the z offset auto homing. I'll make sure it always homes at the set homing point, as well as ensuring it bring the nozzle to that point in order to set the offset. I think this should make the changes to the auto home item unnecessary, as its not specifically just for setting the z offset. Thanks for your work, hopefully I should start getting it merged in the next few days.